modules/components/src/DataTable/DataTable.js

```diff
@@ -77,69 +79,72 @@ class DataTableWithToolbar extends React.Component {
       enableDropDownControls = false,
-      exportTSVText,
-      exportTSVFilename,
       exporter,
-      transformParams,
-      maxPagesOptions,
-      projectId = PROJECT_ID,
-      downloadUrl = urlJoin(ARRANGER_API, projectId, 'download'),
-      onSortedChange = () => {},
-      alwaysSorted = [],
+      exportTSVFilename,
+      exportTSVText,
+      fetchData,
+      filterInputPlaceholder,
       initalSelectedTableRows,
+      InputComponent,
       keepSelectedOnPageChange = false, // If false, this will reset the selection to empty on reload even if sessionStorage is enabled. To keep selections after reload set this to true.
+      loading = null,
+      maxPagesOptions,
+      onColumnsChange = () => {},
+      onFilterChange = () => {},
+      onMultipleColumnsChange = () => {},
+      onSortedChange = () => {},
+      projectId = PROJECT_ID,
+      selectedTableRows = [],
+      setSelectedTableRows = () => {},
       showFilterInput = true,
-      filterInputPlaceholder,
-      InputComponent,
-      customHeaderContent = null,
+      sqon,
+      tableStyle,
+      toolbarStyle,
+      transformParams,
     } = this.props;
     const { page, pageSize, sorted, total } = this.state;
+
     return (
       <>
         <TableToolbar
-          filterInputPlaceholder={filterInputPlaceholder}
-          onFilterChange={onFilterChange}
-          style={toolbarStyle}
-          propsData={data}
-          customActions={customActions}
-          allowTogglingColumns={allowTogglingColumns}
+          allColumns={config.allColumns}
           allowTSVExport={allowTSVExport}
-          sqon={sqon}
+          allowTogglingColumns={allowTogglingColumns}
+          columnDropdownText={columnDropdownText}
           columns={config.columns}
+          customActions={customActions}
+          customHeaderContent={customHeaderContent}
           defaultColumns={config.defaultColumns}
-          onColumnsChange={onColumnsChange}
-          onMultipleColumnsChange={onMultipleColumnsChange}
-          total={total}
-          page={page}
-          pageSize={pageSize}
-          type={config.type}
-          columnDropdownText={columnDropdownText}
+          downloadUrl={urlJoin(ARRANGER_API, projectId, 'download')}
           enableDropDownControls={enableDropDownControls}
-          exportTSVText={exportTSVText}
           exportTSVFilename={exportTSVFilename}
+          exportTSVText={exportTSVText}
           exporter={exporter}
-          transformParams={transformParams}
-          downloadUrl={downloadUrl}
+          filterInputPlaceholder={filterInputPlaceholder}
           InputComponent={InputComponent}
+          keyField={config.keyField}
+          onColumnsChange={onColumnsChange}
+          onFilterChange={onFilterChange}
+          onMultipleColumnsChange={onMultipleColumnsChange}
+          page={page}
+          pageSize={pageSize}
+          propsData={data}
+          selectedTableRows={selectedTableRows}
```

Comment:
Pass the selectedTableRows to the toolbar. 
Sort the other props alphabetically.